constraint/pkg/apis/constraints/apis.go

```diff
@@ -45,3 +56,123 @@ func GetEnforcementAction(constraint *unstructured.Unstructured) (string, error)
 
 	return action, nil
 }
+
+func IsEnforcementActionScoped(action string) bool {
+	return action == EnforcementActionScoped
+}
+
+// GetEnforcementActionsForEP returns a map of enforcement actions for enforcement points.
+func GetEnforcementActionsForEP(constraint *unstructured.Unstructured, eps []string) (map[string]map[string]bool, error) {
+	actionsForEPs := make(map[string]map[string]bool)
+
+	for _, enforcementPoint := range eps {
+		actionsForEPs[enforcementPoint] = make(map[string]bool)
+	}
+
+	// Access the scopedEnforcementAction field
+	scopedActions, found, err := getNestedFieldAsArray(constraint.Object, "spec", "scopedEnforcementActions")
```

Comment:
It might be helpful to simply define a `ScopedEnforcementActions` struct and convert the JSON to the struct via DefaultUnstructuredConverter (example):

https://github.com/open-policy-agent/frameworks/blob/5446e9edab0b7c9177335e6ef0e286e2a435ff4f/constraint/pkg/client/drivers/k8scel/transform/make_vap_objects.go#L113

That way:

* we have a place to point to for the expected schema of scopedEnforcementActions
* You don't need to deal with getNested<thing> commands/error processing/downstream conversions